### DISCUSSION TITLE
How to cache json response of html pages - PWA / Offline-first

### DISCUSSION THREAD

**Discussion by smontlouis** [vercel/next.js#22536] [Help] [answered]
Let's say I generated the following static pages:

```
Module A
    Page 1
    Page 2
    Page 3

Module B
    Page 1
    Page 2
    Page 3

Module C
    Page 1
    Page 2
    Page 3

```

The user is able to store any module offline by using the Cache API by clicking on a download button. [See more here](https://web.dev/offline-cookbook/#on-user-interaction)

Ideally I planned to store the url like the following:

```
cache.addAll(['/module-a/page-1', '/module-b/page-2', '/module-a/page-3'])
```

But those are only useful for hard reload, because NextJS is using the json version of those pages to be able to make an ajax request. So something like that would be better:

```
cache.addAll([
  '/_next/data/HZ8M8zyMCFRsBbtbu1fq2/module-a/page-1.json',
  '/_next/data/HZ8M8zyMCFRsBbtbu1fq2/module-a/page-2.json',
  '/_next/data/HZ8M8zyMCFRsBbtbu1fq2/module-a/page-3.json',
])
```

I don't know how to retrieve the latter urls in order to cache them.
Is there a way to find this generated id string ?

Is there another way to do what I want to achieve ?

**Comment by stigkj** [comment, marked as answer]
I have not tried, but maybe this is build id that the [`generateBuildId`](https://nextjs.org/docs/api-reference/next.config.js/configuring-the-build-id) in `next.config.js` is used for? You could try something like this to let the id be the git hash of HEAD (using [next-build-id](https://github.com/nexdrew/next-build-id)):
```
const nextBuildId = require('next-build-id')
module.exports = {
  generateBuildId: () => nextBuildId({ dir: __dirname })
}
```

I would also think you need to cache both the regular page and the json? Otherwise, a reload would not be cached.

**Reply by smontlouis** [reply]
Indeed ! Thank you !

Do you know if there is an issue keeping the build id static ?

**Reply by stigkj** [reply]
Do not think a static build id is a good idea, but not sure what the problem would be. Maybe you risk getting the previous deployment's data?
